types/options.d.ts

```diff
@@ -144,7 +144,7 @@ export interface RenderContext<Props=DefaultProps> {
   injections: any
 }
 
-export type Prop<T> = { (): T } | { new(...args: any[]): T & object }
+export type Prop<T> = { (): T } | { new(...args: any[]): T & object } | {new(...args: string[]): Function;}
```

Comment:
Please unify the code style with existing one 🙂 
`{ new(...args: string[]): Function }`